ui/components/app/nfts-tab/nfts-tab.js

```diff
@@ -66,6 +71,37 @@ export default function NftsTab() {
   const showNftBanner = hasAnyNfts === false;
   const currentNetwork = useSelector(getCurrentNetwork);
   const currentLocale = useSelector(getCurrentLocale);
+  const hasAnyPreviouslyOwnedNfts = previouslyOwnedCollection.nfts.length > 0;
+  useEffect(() => {
+    if (
+      (isIpfsEnabled || openSeaEnabled) &&
+      (hasAnyNfts || hasAnyPreviouslyOwnedNfts)
+    ) {
+      const allNfts = [...previouslyOwnedCollection.nfts];
+      for (const key in collections) {
+        if (Object.hasOwn(collections, key)) {
+          allNfts.push(...collections[key].nfts);
+        }
+      }
+      const nftsToUpdateMetadata = allNfts.filter(
+        (singleNft) =>
+          !singleNft.name && !singleNft.description && !singleNft.image,
+      );
```

Comment:
Try this to simplify the collecting for `allNfts` if you think it is a good approach :)
```
const allNfts = previouslyOwnedCollection.nfts.concat(
  Object.values(collections).flatMap(collection => collection.nfts)
);
```